### PR TITLE
Add initial support for syntax proposal

### PR DESCRIPTION
**Pull request by kristoferbaxter**
Adds support for resolving `syntax: esmodules` when specifying the option.

### CODE REVIEW

**Comment by kristoferbaxter** [comment]
This PR will be ready to review following:

- [ ] Tests mirroring `browser` field
- [ ] Blog post explaining the `syntax` proposal for packages on npm/yarn.

@developit

File: package.json [hunk 1]
```diff
@@ -6,8 +6,8 @@
     "buble": "^0.19.7",
     "es5-ext": "^0.10.49",
     "eslint": "^5.16.0",
-    "mocha": "^6.1.2",
-    "rollup": "^1.9.3",
+    "mocha": "^6.1.3",
+    "rollup": "^1.10.1",
```
Comment (kristoferbaxter) [inline package.json]: These changes were done by `npm audit fix`, I can back them out of this PR.

File: .eslintrc [hunk 2]
```diff
@@ -20,7 +20,7 @@
     },
     "extends": "eslint:recommended",
     "parserOptions": {
-        "ecmaVersion": 6,
+        "ecmaVersion": 2017,
```
Comment (kristoferbaxter) [inline .eslintrc]: This PR introduces a test using `async`/`await` which necessitated changing the parserOptions for ESLint.

File: src/index.js [hunk 3]
```diff
@@ -41,7 +41,10 @@ function getMainFields (options) {
 	let mainFields;
 	if (options.mainFields) {
 		if ('module' in options || 'main' in options || 'jsnext' in options) {
-			throw new Error(`node-resolve: do not use deprecated 'module', 'main', 'jsnext' options with 'mainFields'`);
+			throw new Error(`node-resolve: do not use deprecated 'module', 'main', or 'jsnext' options with 'mainFields'`);
```
Comment (kristoferbaxter) [inline src/index.js]: This is a minor wording change, and can be removed.

**Comment by kristoferbaxter** [comment]
@lukastaegert – This PR is failing on Travis since it's testing with Node 6 (which has excited LTS support). https://nodejs.org/en/about/releases/

Do you intend to keep support Node 6? If so I can refactor the test using an `async function` back to a `function:Promise` without `await` keywords.

**Comment by lukastaegert** [comment]
Yes, I think we will still stick with Node 6 for some time. Not too long ago, I even got complaints because I accidentally broke Node 4 compatibility in one plugin. Maybe next year we can think about Node 8...

**Comment by kristoferbaxter** [comment]
No problem. I'll move this method back and restore the Buble pipeline.

Thanks for the info!

**Comment by kristoferbaxter** [comment]
I've resolved this issue, the PR is ready for review.

**Comment by lukastaegert** [comment]
What about the blog post? Would be sad to support something that nobody knows about.

**Comment by kristoferbaxter** [comment]
Definately, we're all a bit slow right now in recovery from Google IO. We'll get the post ready for next week.

**Comment by shellscape** [comment]
@kristoferbaxter I'd hate to close this as abandoned after all the work that was put into it. It looks like there are now quite a few conflicts since your last commit. Would you be interested in updating this PR? We're getting ready to move this plugin to https://github.com/rollup/plugins and are tying up loose ends.

**Comment by kristoferbaxter** [comment]
Let's go ahead and close it. There doesn't appear to be consensus on this direction.

**Comment by shellscape** [comment]
Thank you for following up.

**Pull request closed without merging**